lib/ansible/modules/clustering/kubevirt/k8s_service.py

```diff
@@ -2,64 +2,113 @@
 # -*- coding: utf-8 -*-
 
 # Copyright (c) 2018, KubeVirt Team <@kubevirt>
-# Apache License, Version 2.0
-# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
 
 ANSIBLE_METADATA = {'metadata_version': '1.1',
                     'status': ['preview'],
                     'supported_by': 'community'}
 
 DOCUMENTATION = '''
----
+
 module: k8s_service
 
 short_description: Manage Services on Kubernetes
 
-description:
-    - Use Openshift Python SDK to manage Services on Kubernetes
-
 version_added: "2.8"
 
 author: KubeVirt Team (@kubevirt)
 
-options:
-    name:
-        description:
-            - Use to specify a Service object name.
-        required: true
-        type: str
-    namespace:
-        description:
-            - Use to specify a Service object namespace.
-        required: true
-        type: str
-    type:
-        description:
-            - Specifies the type of Service to create.
-            - U(https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types)
-        choices:
-            - NodePort
-            - ClusterIP
-            - LoadBalancer
-            - ExternalName
-        default: ClusterIP
-        required: true
-    ports:
-        description:
-            - A list of ports to expose.
-            - U(https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services)
-        required: true
-        type: list
-    selector:
-        description:
-            - Label selectors identify objects this Service should apply to.
-            - U(https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)
-        required: true
-        type: dict
+description:
+  - Use Openshift Python SDK to manage Services on Kubernetes
 
 extends_documentation_fragment:
   - k8s_auth_options
-  - k8s_resource_options
+
+options:
+  resource_definition:
+    description:
+    - A partial YAML definition of the Service object being created/updated. Here you can define Kubernetes
```

Comment:
Any reason to not use doc from `k8s_state_options`?

Reply:
Three:
1. The description isn't 100% fitting.
2. If you need `src`, you should just use `k8s` module.
3. By default `resource_definition` supports both a single object and a list of objects. I'm enforcing only the former.

In general the approach is that this gets merged without 100% code reuse, but once another k8s_something module gets added, that'll be the time to extract some more common code and generalize stuff. The current "common" code would be misleading and not 100% correct for this particular module.